File: widgetssdk/src/main/java/com/glia/widgets/core/dialog/domain/SetOverlayPermissionRequestDialogShownUseCase.java
package com.glia.widgets.core.dialog.domain;

import com.glia.widgets.core.dialog.PermissionDialogManager;

public class SetOverlayPermissionRequestDialogShownUseCase {
    private final PermissionDialogManager permissionDialogManager;

    public SetOverlayPermissionRequestDialogShownUseCase(PermissionDialogManager permissionDialogManager) {
        this.permissionDialogManager = permissionDialogManager;
    }

    public void execute() {
        this.permissionDialogManager.setOverlayPermissionDialogShown();
    }
}

File: widgetssdk/src/main/java/com/glia/widgets/core/dialog/domain/SetOverlayPermissionRequestDialogShownUseCase.kt
package com.glia.widgets.core.dialog.domain

import com.glia.widgets.core.dialog.PermissionDialogManager

internal interface SetOverlayPermissionRequestDialogShownUseCase {
    operator fun invoke()
}

internal class SetOverlayPermissionRequestDialogShownUseCaseImpl(private val permissionDialogManager: PermissionDialogManager) :
    SetOverlayPermissionRequestDialogShownUseCase {
    override fun invoke() = permissionDialogManager.setOverlayPermissionDialogShown()
}
